.kiro/specs/auterity-ai-platform-unified/templates/compliance-aware-agent-template.py

```
@@ -0,0 +1,676 @@
+                    result = await self._execute_ai_task(task, monitor)
+                else:
+                    result = await self._execute_standard_task(task, monitor)
+                
+                execution_time = (datetime.utcnow() - start_time).total_seconds()
+                result.execution_time = execution_time
+                
+                return result
+                
+        except asyncio.TimeoutError:
+            await monitor.log_timeout()
+            raise Exception(f"Task {task.id} exceeded time limit of {task.timeout_seconds} seconds")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)